src/foam/dao/RestDAO.js

```diff
@@ -87,6 +88,17 @@ foam.CLASS({
        * Each key's value is network-foam-jsonified.
        */
       var query = [];
+
+      // Proxy non-serializable sinks (with either "sink" or an ArraySink).
+      // When "sink" is serializable, make "proxySink" falsey as a signal to:
+      // (1) Send the serializable sink (" ! proxySink" below);
+      // (2) Signal to "onSelectResponse" that sink in payload should be passed
+      //     back to caller.
+      var proxySink = this.Serializable.isInstance(sink) ? null :
```

Comment:
This works. Needed to pass along `sink || this.ArraySink.create()` to `onSelectResponse`, but that's not problem.